GitUI/UserControls/FileStatusDiffCalculator.cs

```diff
@@ -4,41 +4,44 @@
 
 namespace GitUI
 {
-    public sealed partial class FileStatusDiffCalculator
+    public sealed partial class FileStatusDiffCalculator(Func<IGitModule> getModule)
     {
-        private readonly Func<IGitModule> _getModule;
-
         // Currently bound revisions etc. Cache so we can reload the view, if AppSettings.ShowDiffForAllParents is changed.
         private FileStatusDiffCalculatorInfo _fileStatusDiffCalculatorInfo = new();
 
         // Default helper functions, can be set
         public Func<ObjectId, string>? DescribeRevision { get; set; }
         public Func<GitRevision, GitRevision>? GetActualRevision { get; set; }
 
-        public FileStatusDiffCalculator(Func<IGitModule> getModule)
-        {
-            _getModule = getModule;
-        }
-
-        public IReadOnlyList<FileStatusWithDescription> Reload()
-            => SetDiffs(_fileStatusDiffCalculatorInfo.Revisions,
-                _fileStatusDiffCalculatorInfo.HeadId, _fileStatusDiffCalculatorInfo.AllowMultiDiff, cancellationToken: default);
-
-        public IReadOnlyList<FileStatusWithDescription> SetDiffs(
+        public void SetDiff(
             IReadOnlyList<GitRevision> revisions,
             ObjectId? headId,
-            bool allowMultiDiff,
-            CancellationToken cancellationToken)
+            bool allowMultiDiff)
         {
             _fileStatusDiffCalculatorInfo.Revisions = revisions;
             _fileStatusDiffCalculatorInfo.HeadId = headId;
             _fileStatusDiffCalculatorInfo.AllowMultiDiff = allowMultiDiff;
+        }
 
-            if (revisions?.Count is not > 0 || revisions[0] is not GitRevision selectedRev)
+        public IReadOnlyList<FileStatusWithDescription> Calculate(CancellationToken cancellationToken = default)
+        {
+            if (_fileStatusDiffCalculatorInfo.Revisions?.Count is not > 0
+                || _fileStatusDiffCalculatorInfo.Revisions[0] is not GitRevision selectedRev)
             {
-                return Array.Empty<FileStatusWithDescription>();
+                return Array.Empty<FileStatusWithDescription>().ToList();
```

Comment:
❓ Why?

Reply:
A consequence of backporting from another branch where this was done in a different way. Reverting.

Reply:
reverted